### ISSUE TITLE
Setup documentation tooling (sphinx, sphinx-needs)

### ISSUE THREAD

**Issue by ltekieli** [eclipse-score/score#78]
* Introduce initial highlighting, preview and linting for .rst and .md files (-> esbonio) -> covered by #30
* Introduce initial code completion for .rst and .md files (-> templates and/or sphinx-needs extension) -> covered by 30
* Introduce first checks like mandatory attributes (-> pydantic based metamodel) < partially blocked by #39 (expected Jan 2025). But we can start with the basics etc.
* Enable drawing images (-> plantuml, mermaid and/or drawio)
* Extend meta model with first few elements
* Initial support for the first multi-repo use cases

**Comment by AlexanderLanin** [comment]
Consider including tool requirements in this ticket

**Comment by AlexanderLanin** [comment]
Updated list:
[x] Introduce initial highlighting, preview and linting for .rst and .md files (-> esbonio) (clone of #30)
[ ] Introduce initial code completion for .rst and .md files (-> templates and/or sphinx-needs extension) (clone of #30)
[ ] Introduce first checks like mandatory attributes (-> pydantic based metamodel)
[ ] Enable drawing images (-> plantuml, mermaid and/or drawio)
[ ] Extend meta model with first few elements

out of scope:
* Initial support for the first multi-repo use cases

**Comment by AlexanderLanin** [comment]
This is at least 95% done. We'll open new dedicated issues for whatever is remaining.
